great_expectations/checkpoint/actions.py

```diff
@@ -128,12 +160,30 @@ class (for open source Great Expectations) or the GeCloudIdentifier (from Great
         """
         return NotImplementedError
 
+    @staticmethod
+    def _build_renderer(config: dict) -> Renderer:
+        renderer = instantiate_class_from_config(
+            config=config,
+            runtime_environment={},
+            config_defaults={},
+        )
+        if not renderer:
+            raise ClassInstantiationError(
+                module_name=config.get("module_name"),
+                package_name=None,
+                class_name=config.get("class_name"),
+            )
+        return renderer
+
 
 class DataDocsAction(ValidationAction):
-    def __init__(self) -> None:
+    _build_data_docs: Callable = PrivateAttr()
+    _get_docs_sites_urls: Callable = PrivateAttr()
+
+    def __init__(self, **data: Any) -> None:
         from great_expectations import project_manager
 
-        super().__init__()
+        super().__init__(**data)
         self._build_data_docs = project_manager.build_data_docs
         self._get_docs_sites_urls = project_manager.get_docs_sites_urls
```

Comment:
Is this the pattern for Pydantic private attrs? Feels a bit bad to override init?

Reply:
Note this also ruins autocomplete in editors

Reply:
Both of these classes are parent classes and won't be exposed to the end user during normal workflows